actions/workflows/st2_pkg_upgrade_e2e_test.yaml

```diff
@@ -161,6 +170,10 @@ tasks:
               distro: <% result().output.distro %>
               versions: <% result().output.versions %>
               version_str: <% result().output.versions.items().select( $[0] + "=" + $[1]).join("\n\t") %>
+          # determine the list of packages to upgrade base on OS
+          # modern OSes (el8+, u18+) do not have mistral and that is our default
+          # we have exceptions for the "legacy" cases as they will drop off with time
+          - pkg_list: <% switch(ctx().distro.endsWith('6') => ctx().pkg_list_legacy, ctx().distro.endsWith('7') => ctx().pkg_list_legacy, true => ctx().pkg_list_modern)  %>
```

Comment:
This makes it impossible to grep for `el6`, `el7`, or `u16` when we're trying to remove those OSes, and pretty much guarantees us a bug in the future. Can you make these explicitly check equality of `ctx().distro` with those legacy OSes?